networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AggregateGossipManager.java

```diff
@@ -37,26 +35,26 @@ public AggregateGossipManager(
       final GossipEncoding gossipEncoding,
       final ForkInfo forkInfo,
       final SignedAggregateAndProofValidator validator,
-      final EventBus eventBus) {
+      final GossipedAttestationConsumer gossipedAttestationConsumer) {
     this.gossipEncoding = gossipEncoding;
-
-    final AggregateTopicHandler aggregateTopicHandler =
-        new AggregateTopicHandler(gossipEncoding, forkInfo, validator, eventBus);
-    this.channel = gossipNetwork.subscribe(aggregateTopicHandler.getTopic(), aggregateTopicHandler);
-
-    this.eventBus = eventBus;
-    eventBus.register(this);
+    final AggregateAttestationTopicHandler aggregateAttestationTopicHandler =
+        new AggregateAttestationTopicHandler(
+            gossipEncoding, forkInfo, validator, gossipedAttestationConsumer);
+    this.channel =
+        gossipNetwork.subscribe(
+            aggregateAttestationTopicHandler.getTopic(), aggregateAttestationTopicHandler);
   }
 
-  @Subscribe
-  public void onNewAggregate(final SignedAggregateAndProof aggregateAndProof) {
-    final Bytes data = gossipEncoding.encode(aggregateAndProof);
+  public void onNewAggregate(final ValidateableAttestation validateableAttestation) {
+    if (validateableAttestation.isGossiped() || !validateableAttestation.isAggregate()) {
```

Comment:
```suggestion
    if (!validateableAttestation.isAggregate() || !validateableAttestation.markGossiped()) {
```
as per comments below.

Reply:
Done.